src/adapter/etl-adapter-google-sheet/src/Flow/ETL/Adapter/GoogleSheet/GoogleSheetExtractor.php

```diff
@@ -50,20 +50,31 @@ public function extract(FlowContext $context) : \Generator
          */
         $values = $response->getValues() ?? [];
 
+        $totalRows = 0;
+
         if ($this->withHeader && [] !== $values) {
-            /** @var array<string> $headers */
-            $headers = $values[0];
-            unset($values[0]);
-            $totalRows = 1;
-        } else {
-            $totalRows = 0;
+            foreach ($values as $index => $row) {
+                if ([] === $row) {
+                    // Remove empty rows at the beginning of a sheet
+                    unset($values[$index]);
```

Comment:
hmm wont that remove any empty row, regardless of the position in the sheet? Like for example one in the middle?

Reply:
No cause there is a `break;` after assign of first found data, so it will remove just first rows till any data found.

Reply:
oh right 👍